src/layer.js

```diff
@@ -216,4 +228,8 @@ Layer.prototype.draw = function(data) {
 			}
 		}
 	}
+
+	if (!selection.empty()) {
```

Comment:
What's the reasoning behind calling remove on the selection if it's not empty? I might be missing some context here.